run/admin_dash.py

```diff
@@ -0,0 +1,225 @@
+import streamlit as st
+from dotenv import load_dotenv
+from sqlalchemy import Date, cast, select
+
+from noshow.dashboard.connection import init_session
+from noshow.database.models import (
+    ApiCallResponse,
+    ApiPatient,
+    ApiPrediction,
+    ApiRequest,
+)
+from noshow.preprocessing.utils import add_working_days
+
+load_dotenv()
+
+date_3_days = add_working_days(datetime.today(), 3)
+
+# Global and env variables
+db_user = os.environ["DB_USER"]
+db_passwd = os.environ["DB_PASSWD"]
+db_host = os.environ["DB_HOST"]
+db_port = os.environ["DB_PORT"]
+db_database = os.environ["DB_DATABASE"]
+
+
+def kpi_page():
+    """Page that contains basic KPIs for the no-show project"""
+    st.write("## KPIs")
+
+    if len(date_input) != 2:
+        st.info("Selecteer een tijdsperiode")
+        return
+
+    with session_object() as session:
+        call_results = session.execute(
+            select(
+                ApiPrediction.start_time,
+                ApiPrediction.clinic_name,
+                ApiCallResponse.call_status,
+                ApiCallResponse.call_outcome,
+            )
+            .outerjoin(ApiPrediction.callresponse_relation)
+            .outerjoin(ApiPrediction.patient_relation)
+            .where(
+                (cast(ApiPrediction.start_time, Date) >= date_input[0])
+                & (cast(ApiPrediction.start_time, Date) <= date_input[1])
+                & (ApiPatient.treatment_group == 1)
+            )
+        ).all()
+
+    if not call_results:
+        st.info("Geen data beschikbaar voor deze periode")
+        return
+
+    call_results_df = pd.DataFrame(call_results)
+    call_results_df["date"] = call_results_df["start_time"].dt.date
+    call_results_df.loc[call_results_df["call_outcome"].isnull(), "call_outcome"] = (
+        "Niet gebeld"
+    )
+    call_results_df.loc[call_results_df["call_outcome"] == "Geen", "call_outcome"] = (
+        "Onbereikbaar"
+    )
+    call_results_df["color_sort"] = call_results_df["call_outcome"].map(
+        {
+            "Niet gebeld": 0,
+            "Onbereikbaar": 1,
+            "Herinnerd": 2,
+            "Verzet/Geannuleerd": 3,
+            "Bel me niet": 4,
+        }
+    )
+
+    metric_cols = st.columns(4)
+    metric_cols[0].metric(
+        "Aantal Hoofdagenda's", call_results_df["clinic_name"].nunique()
+    )
+    metric_cols[1].metric(
+        "Aantal patienten herinnerd",
+        len(call_results_df[call_results_df["call_outcome"] == "Herinnerd"]),
+    )
+    metric_cols[2].metric(
+        "Aantal afspraken verzet of geannuleerd",
+        len(call_results_df[call_results_df["call_outcome"] == "Verzet/Geannuleerd"]),
+    )
+    percentage_reached = (
+        len(
+            call_results_df[
+                call_results_df["call_outcome"].isin(
+                    ["Herinnerd", "Verzet/Geannuleerd"]
+                )
+            ]
+        )
+        / len(call_results_df)
+        * 100
+    )
+    metric_cols[3].metric(
```

Comment:
Maybe also nice to have the metric 'percentage patient gebeld'. This metric can be useful to monitor if we need to deploy more students